app/database/accessor/breeze_buddy/blacklisted_numbers.py

```
@@ -0,0 +1,204 @@
+
+
+async def _invalidate_cache(
+    phone_number: str, merchant_id: Optional[str] = None
+) -> None:
+    """Invalidate Redis cache entries for a phone number."""
+    try:
+        redis = await get_redis_service()
+        # Delete both the specific merchant key and the general key
+        await redis.delete(_cache_key(phone_number, merchant_id))
+        await redis.delete(_cache_key(phone_number))
+    except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)